spec/roles/planned_timetable_slot_spec.rb

```diff
@@ -7,23 +7,34 @@
 describe PlannedTimetableSlot do
 
   let(:slot) { Fabricate(:timetable_slot, first_hour: 1, duration: 2, parity: 'both', day: :monday) }
+  let(:slot_with_weeks_and_times) { double(id: 1, first_hour: 1, duration: 2, parity: nil, day: 1, start_time: Time.parse('1901-01-01 07:30:00'), end_time: Time.parse('1901-01-01 08:10:00'), weeks: [(1..1), (3..3)]) }
   let(:period) { Period.parse('7:30', '9:00') }
-  let(:converter) { instance_double(Sirius::TimeConverter, convert_time: period) }
-  let(:semester_calendar) { instance_double(PlannedSemesterPeriod, plan: [period]) }
+  let(:semester_calendar) { PlannedSemesterPeriod.new(Fabricate(:teaching_semester_period)) }
   let(:faculty_semester) { Fabricate.build(:faculty_semester) }
+  let(:converter) { Sirius::TimeConverter.new(hour_starts: faculty_semester.hour_starts, hour_length: faculty_semester.hour_duration) }
+  let(:weeks_dates) {[[Date.parse("2014-09-22"), Date.parse("2014-09-28")], [Date.parse("2014-09-29"), Date.parse("2014-10-05")], [Date.parse("2014-10-06"), Date.parse("2014-10-12")]]}
   subject(:planned_slot) { described_class.new(slot, converter) }
+  subject(:planned_slot_with_weeks_and_times) { described_class.new(slot_with_weeks_and_times, converter) }
```

Comment:
RSpec uses `context` (or `describe`) blocks for declaring different *context*s (inputs) of the tests.

See https://github.com/cvut/sirius/commit/0c54d47ce6037efff617c7605ebd2d7fb8a3acc4#diff-95f6317ac00c7e6c68dee55b9e7b2757 (without `weeks_dates` in this case).